OpenRA.Renderer.Sdl2/Sdl2GraphicsDevice.cs

```diff
@@ -330,6 +330,39 @@ public void SetLineWidth(float width)
 			ErrorHandler.CheckGlError();
 		}
 
+		public Bitmap TakeScreenshot()
+		{
+			var rect = new Rectangle(Point.Empty, size);
+			var bitmap = new Bitmap(rect.Width, rect.Height, System.Drawing.Imaging.PixelFormat.Format32bppArgb);
+			var data = bitmap.LockBits(rect,
+				System.Drawing.Imaging.ImageLockMode.WriteOnly, System.Drawing.Imaging.PixelFormat.Format32bppArgb);
+
+			try
+			{
+				int packAlignment;
+
+				unsafe
+				{
+					GL.GetInteger(GetPName.PackAlignment, out packAlignment);
+				}
+
+				GL.PixelStore(PixelStoreParameter.PackAlignment, 1);
+
+				GL.ReadPixels(rect.X, rect.Y, rect.Width, rect.Height, PixelFormat.Bgra, PixelType.UnsignedByte, data.Scan0);
```

Comment:
Maybe a dumb question - what is the criterion for using the `unsafe { }` block? Both `GetInteger` and `ReadPixels` return data through pointers...